src/libraries/System.Text.Json/tests/Serialization/CollectionTests/CollectionTests.Dictionary.cs

```diff
@@ -249,6 +249,36 @@ public static void ImplementsDictionary_DictionaryOfString()
                 json = JsonSerializer.Serialize<object>(obj);
                 Assert.Equal(JsonString, json);
             }
+
+            {
+                GenericStructIDictionaryWrapper<string, string> obj = JsonSerializer.Deserialize<GenericStructIDictionaryWrapper<string, string>>(JsonString);
+                Assert.Equal("World", obj["Hello"]);
+                Assert.Equal("World2", obj["Hello2"]);
+
+                string json = JsonSerializer.Serialize(obj);
+                Assert.Equal(JsonString, json);
+
+                json = JsonSerializer.Serialize<object>(obj);
+                Assert.Equal(JsonString, json);
+            }
+
+            {
+                GenericStructIDictionaryWrapper<string, string>? obj = JsonSerializer.Deserialize<GenericStructIDictionaryWrapper<string, string>?>(JsonString);
+                Assert.True(obj.HasValue);
+                Assert.Equal("World", obj.Value["Hello"]);
+                Assert.Equal("World2", obj.Value["Hello2"]);
+
+                string json = JsonSerializer.Serialize(obj);
+                Assert.Equal(JsonString, json);
+
+                json = JsonSerializer.Serialize<object>(obj);
+                Assert.Equal(JsonString, json);
+            }
+
+            {
+                GenericStructIDictionaryWrapper<string, string>? obj = JsonSerializer.Deserialize<GenericStructIDictionaryWrapper<string, string>?>("null");
```

Comment:
What happens when serializing a `default` instance of `GenericStructIDictionaryWrapper<string, string>`? Test this for `GenericStructIListWrapper<int>` as well.